drivers/inc/stm32f446xx_usart_driver_course.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * stm32f446xx_usart_driver_course.h
+ *
+ *  Created on: Apr 21, 2020
+ *      Author: quentin
+ */
+#define QUENTIN
+
+#ifdef QUENTIN
+
+#ifndef INC_STM32F446XX_USART_DRIVER_COURSE_H_
+#define INC_STM32F446XX_USART_DRIVER_COURSE_H_
+
+#include "stm32f446xx_course.h"
+
+/* GPIO_PIN_CONF_TypeDef */
+typedef struct{
+	uint8_t USART_Instance;
+	uint32_t USART_BaudRate; //size might change
+	uint8_t USART_WordLength;
+	uint8_t USART_StopBits;//0=1 bits, 1=0.5 bit, 2=2 bits, 3=1.5 bit
+	uint8_t USART_Parity;
+	uint8_t USART_Mode;
+	uint8_t USART_HardwareFlowCtrl;
+	uint8_t USART_OverSampling;
+}USART_PIN_CONF_TypeDef;
+
+typedef struct{
+	USART_RegDef_t* pUSART; /*holds register addresses*/
+	USART_PIN_CONF_TypeDef USART_Pin_Config; /* holds pins settings */
+}USART_Handler_TypedDef;
+
+/* INSTANCE */
+#define USART_INSTANCE_1			1
+#define USART_INSTANCE_2			2
+#define USART_INSTANCE_3			3
+#define USART_INSTANCE_4			4
+#define USART_INSTANCE_5			5
+#define USART_INSTANCE_6			6
+
```

Comment:
If you already using the USART_RegDef_t, why did you included also the Instance?

The usual reason to do this is to Encapsulate this register, doing:
Typedef  *USART_RegDef_t USART_RegDefine_t
This will make only possible to access the USART_RegDef_t->  in the file, then you need to create a allocation function to return a pointer to this register, this is similar to a Constructor in Cpp